float32 camera_coordinate_x
float32 camera_coordinate_y
